app/views/handlers/block.py

```diff
@@ -44,6 +44,8 @@ def __init__(
         self._return_to = request_args.get("return_to")
         self._return_to_answer_id = request_args.get("return_to_answer_id")
         self._return_to_block_id = request_args.get("return_to_block_id")
+        self._return_to_list_name = request_args.get("return_to_list_name")
```

Comment:
Do we actually need to pass this?. This is deterministic if we have `return_to_list_item_id`. We should avoid adding anything to the URL that we don't strictly need. From a quick scan, the only place we actually need list name is `_get_return_to_for_grand_calculated_summary`, but in there, that can be worked out dynamically as you have access to the list store in router.py. You just need a map in list store for list item id to list name look up.

Reply:
Good point, we don't, with the lookup in the list store it can indeed be removed, so have done that. I went with going through the lists as think it should be ok in this specific case